Formula/cwb3.rb

```diff
@@ -0,0 +1,102 @@
+class Cwb3 < Formula
+  desc "Tools for managing and querying large text corpora with linguistic annotations"
+  homepage "https://cwb.sourceforge.io/"
+  license "GPL-2.0-or-later"
+
+  stable do
+    url "https://downloads.sourceforge.net/project/cwb/cwb/cwb-3.5-RC/cwb-3.4.33-src.tar.gz"
+    sha256 "947d5f6d710fd8f818c2dc027e991b898308d581cb9b89c4ce6db054fbb20948"
+  end
+
+  head do
+    url "svn://svn.code.sf.net/p/cwb/code/cwb/trunk"
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "gettext"
+  depends_on "glib"
+  depends_on "pcre"
+  depends_on "readline"
+
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+  uses_from_macos "ncurses"
+
+  def install
+    system("make", "all",
+      "PLATFORM=homebrew-formula", "SITE=homebrew-formula", "FULL_MESSAGES=1",
+      "PREFIX=#{prefix}", "HOMEBREW_ROOT=#{HOMEBREW_PREFIX}")
+    ENV.deparallelize
+    system("make", "install",
+      "PLATFORM=homebrew-formula", "SITE=homebrew-formula", "FULL_MESSAGES=1",
+      "PREFIX=#{prefix}", "HOMEBREW_ROOT=#{HOMEBREW_PREFIX}")
```

Comment:
```suggestion
    args = %W[
      PLATFORM=homebrew-formula
      SITE=homebrew-formula
      FULL_MESSAGES=1
      PREFIX=#{prefix}
      HOMEBREW_ROOT=#{HOMEBREW_PREFIX}
    ]

    system "make", "all", *args
    ENV.deparallelize
    system "make", "install", *args
```

Reply:
Thanks for looking at this so quickly! I've just been investigating the issue myself.  A quick test run on my MacBook suggests that `-lintl` is no longer needed (it's from pkg-config on MacOS, but we don't actually seem to be using any of the code).  I will have to fix this in upstream and make a new source release after confirming that it's fine without `-lintl`.

**Question:** Should upstream use `pkg-config` to get the compiler and linker flags instead? I was worried that package names might not be the same in Linux, breaking the build.  Upstream also explicitly uses paths `-L $HOMEBREW_PREFIX/opt/...` so installed software doesn't break when upgrading the prerequisite libraries to newer versions.  When installed via formula `cwb3`, will HomeBrew ensure automatically that required versions of the prerequisites will remain available in Cellar?

Reply:
PS: I will need some guidance on how to submit a new pull request after we've resolved all these issues.

Reply:
Using `pkg-config` might be more portable. You _might_ still need `-lintl` to link with `glib` on macOS, though I think macOS `ld` is smart enough to work the recursive dependency out these days.

Are you concerned about `.pc` files referring to `Cellar` paths? That's not an issue on macOS because we rewrite install names to avoid referencing them. I believe it's also not an issue on Linux, but I'm not up-to-speed on the details.

There's no need to submit a new pull request -- you can just modify this one with `git commit --amend` and `git push --force`.

Reply:
Oh right, my bad. It _does_ link against `libintl`, but doesn't require the explicit flag.

There are two concerns about `pkg-config`:

- I don't have Linux Homebrew for testing, so I'm not sure whether `pkg-config libpcre` works there as well, or whether the `.pc` happens to have a different name on some distros. These package names seem to be entirely unsystematic (e.g. `libpcre` vs. `glib-2.0` vs. `ncurses`).
- The paths I get from `pkg-config` on MacOS point to the Cellar rather than opt/, e.g.

```
$ pkg-config --libs glib-2.0
-L/usr/local/Cellar/glib/2.70.2/lib -L/usr/local/opt/gettext/lib -lglib-2.0 -lintl
```

So whenever I upgrade to a new glib version, the Cellar directory is deleted and the installed programs fail to run.  Note that all paths are set in the upstream Makefile, so we'd have to dynamically patch the makefile if we have to rewrite the compiler/linker flags in the formula.

Reply:
`libpcre.pc` is what you get from `./configure && make install`, so `pkg-config libpcre` should work on Linux too. The package names for `pkg-config` should be the same on macOS and Linux (at least for Homebrew).

Does `cwb3` remember the flags from `pkg-config` invocations? As long as it doesn't do so explicitly then the Cellar references shouldn't matter (e.g. try `otool -D $(brew --cellar glib)/*/lib/libglib-2.0.dylib`). Remembering these paths explicitly is atypical -- otherwise we'd need to recompile dependents much more often every time a formula has its `version` or `revision` bumped.

Reply:
CWB simply passes the `-L` flags from `pkg-config` to `gcc` when linking.  This used to cause the described breakage on every `brew upgrade` and hence was changed to manually set paths to opt/.  Perhaps this has changed in the meantime.  Is there any way to install an older/different version of a formula to test whether the build process works with pkg-config now?

Reply:
The functionality of `libintl` on Linux is provided internally by `glibc`, which is automatically linked to all compiled binaries, and should never be explicitly linked with `-lintl`.  On macOS the situation is more complex - I don't believe `libSystem` provides that automatically, so the flag is sometimes needed.

If you want to be able to test the build in Linux locally, you can use the Homebrew docker image we use in CI: ` docker run -it -m 16GB homebrew/ubuntu16.04:master`.  That already has Homebrew set up so the only thing you have to do is check out your branch on there and you can test in the same build environment we use in CI.

Reply:
We run `brew linkage` in CI to check that formulae aren't linking to libraries that no longer exist (this is part of the testing done for dependents when a formula is updated), so this shouldn't be too much of a concern.

> Is there any way to install an older/different version of a formula to test whether the build process works with pkg-config now?

With a bit of `git` hackery, sure. But it's not really straightforward, and probably not worth the time.

Reply:
`pkg-config` indeed seems to work correctly now. Upstream has been fixed to use pkg-config where possible.